constellation_forms/views.py

```diff
@@ -238,11 +238,21 @@ def post(self, request, form_submission_id):
         submission = FormSubmission.objects.get(pk=form_submission_id)
         redirect = reverse('constellation_forms:view_form_submission',
                            args=[form_submission_id])
-        if ("message" not in request.POST or
-                len(request.POST["message"].strip()) == 0):
+
+        if (("message" not in request.POST or
+                len(request.POST["message"].strip()) == 0) and
+                "file" not in request.FILES):
             return HttpResponseRedirect(redirect)
         new_log = Log()
         new_log.message = request.POST["message"]
+        if "file" in request.FILES:
+            new_log.mtype = 4
+            with open(settings.MEDIA_ROOT + request.FILES["file"].name,
```

Comment:
This permits malicious overwriting of files by a user that knows a filename of an existing file.  This should really be formID-logNum-filename to prevent the user being able to fully control the filename.